src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/enchanting/AbstractEnchantmentMachine.java

```diff
@@ -55,4 +64,23 @@ protected void showEnchantmentLevelWarning(@Nonnull BlockMenu menu) {
         menu.replaceExistingItem(22, progressBar);
     }
 
+    protected boolean hasIgnoredLore(@Nonnull ItemStack itemStack) {
+        List<String> ignoredLore = ignoredLores.getValue();
+        if (!useIgnoredLores.getValue()) {
+            return false;
+        }
+        if (itemStack.hasItemMeta()) {
+            ItemMeta itemMeta = itemStack.getItemMeta();
+            if (!itemMeta.hasLore()) {
+                return false;
+            }
+            List<String> itemLore = itemMeta.getLore();
+            for (String lore : ignoredLore) {
+                if (itemLore.contains(ChatColors.color(lore))) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
```

Comment:
This can be tidied up a bit, `ignoredLore` is not used until much further down, so we dont need to declare it this early on.
Also replaced the guard block with a simple if (AND)
```suggestion
        if (useIgnoredLores.getValue() && itemStack.hasItemMeta()) {
            ItemMeta itemMeta = itemStack.getItemMeta();

            if (!itemMeta.hasLore()) {
                return false;
            }

            List<String> itemLore = itemMeta.getLore();
            List<String> ignoredLore = ignoredLores.getValue();

            for (String lore : ignoredLore) {
                if (itemLore.contains(ChatColors.color(lore))) {
                    return true;
                }
            }
        }

        return false;
    }
```

Reply:
Should be edited after I added some check earlier... But I forgot.